util/logtools/splitter.ts

```diff
@@ -20,9 +26,33 @@ export default class Splitter {
     private notifier: Notifier,
     private includeGlobals: boolean,
   ) {
-    // Remap logDefinitions from log type (instead of name) to definition.
-    for (const def of Object.values(logDefinitions))
+    for (const def of Object.values(logDefinitions) as LogDefinition[]) {
```

Comment:
Is there a way to avoid `as LogDefinition[]` here?

Reply:
I tried, but I think we need the forced typing because possibleRsvFields is an optional property, and TS throws a tantrum if the def object doesn't have an explicit type.  I thought the forced typing would be okay in this circumstance because the object values are imported and already of that type (from netdefs) so there's not really any room for whoopsies.   But let me know if you have another idea in mind.

Reply:
```typescript
    const defs: LogDefinitionMap = logDefinitions;
    for (const def of Object.values(defs)) {
```

This will do it. The issue as you say is that because it's an optional property and not every value has this property, the intersection of all of these types does not include it. One option is to add it a non-optional member (that could be undefined, but present). The other option is to more loosely type it as the above.